File: MoonbeltEventManager/Models/Party.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace MoonbeltEventManager.Models
{
    public partial class Party
    {
        public Party()
        {
            PersonParties = new HashSet<PersonParty>();
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public string Venue { get; set; }
        public DateTime? StartDateTime { get; set; }
        public DateTime? EndDateTime { get; set; }

        public virtual ICollection<PersonParty> PersonParties { get; set; }
    }
}

File: MoonbeltPartyManagerApp/src/app/models/party.ts
import { jsonObject, jsonMember } from '../../../node_modules/typedjson';

@jsonObject
export class Party {
    @jsonMember({ name: "id", constructor: Number })
    id!: number;

    @jsonMember({ name: "name", constructor: String })
    title!: string;

    @jsonMember({ name: "venue", constructor: String })
    venue!: string;

    @jsonMember({ name: "startDateTime", constructor: String })
    startDateTime!: string;

    @jsonMember({ name: "endDateTime", constructor: String })
    endDateTime!: string;

   public constructor(init?: Partial<Party>)
   {
      Object.assign(this,init);
   }
}
